### PR TITLE
Singlemonster

### LINKED ISSUE #3: Create Listcomponent
## Value statement - a requirement, not a solution
As a **Player** 
I benefit from **additional Information displayed in the List**
to **get an correct idea of that monster** quicker.

## Description (Text, Scribble, Wireframe, Design)
Bildschirmfoto 2020-08-31 um 06 43 52

## Acceptance criteria
- [x] Display the name, type and challenge-rating of every monster in the component of the list

## Tasks
- [x] Define a design of a list-component in Storybook.
- [x] Build a component with interchangeable text for monstername, type and challenge-rating.
- [x] Fetch the additional information to every monster.
- [x] Insert the additional information from the fetch in the component
- [x] Write  a Unittest

## Size
(large)

### PR DESCRIPTION
**Pull request by janhenrikfock**
Fetching and storing additional information for every component on the dashboard. Then displaying it via the new component "OneMonsterItem".

### CODE REVIEW

File: src/components/AllMonstersList.js [hunk 1]
```diff
@@ -1,41 +1,52 @@
...
+    setMonsterDetails(JSON.parse(localStorage.getItem('monsterdetails')))
+  }
+  console.log(monsterDetails)
   return (
     <>
       <HeadlineStyled>Monsters of DnD</HeadlineStyled>
-      {monsterList.map((monster, index) => (
-        <MonsterEntryStyled key={index}>{monster.name}</MonsterEntryStyled>
+      {monsterDetails.map((monster, index) => (
+        <OneMonsterItem key={index} monster={monster}></OneMonsterItem>
```
Comment (mpagels) [inline src/components/AllMonstersList.js]: index is not a good practice to use it as a key. You can use in your case the monster name, if unique.

File: src/components/AllMonstersList.js [hunk 2]
```diff
@@ -1,41 +1,52 @@
...
 
   useEffect(() => {
-    const loadedArray = JSON.parse(localStorage.getItem('monsternames'))
+    const locallyLoadedMonsterNames = JSON.parse(
+      localStorage.getItem('monsternames')
+    )
 
-    if (!loadedArray || loadedArray.length === 0) {
+    if (!locallyLoadedMonsterNames || locallyLoadedMonsterNames.length === 0) {
       fetch(`https://www.dnd5eapi.co/api/monsters`)
```
Comment (mpagels) [inline src/components/AllMonstersList.js]: no reason to use template literals here

File: src/components/OneMonsterItem.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import React from 'react'
+import styled from 'styled-components'
+
+export default function OneMonsterItem({ monster }) {
+  return (
+    <ContainerStyled>
+      <MonsterNameStyled>{monster.name}</MonsterNameStyled>
+      <ContainerTypeCR>
```
Comment (mpagels) [inline src/components/OneMonsterItem.js:8]: what means CR in ContainerTypye CR?

**Review by mpagels — COMMENTED** [review]
Good Job,
found a few little things and have one question: You named your component AllMonsterList <- LIST, but using not once the corresponding semantic tag in your code. Is this intentional?

File: src/components/AllMonstersList.js [hunk 4]
```diff
@@ -1,41 +1,52 @@
...
-      setMonsterList(loadedArray)
     }
   }, [])
 
+  if (monsterDetails.length !== 0) {
+    saveLocally('monsterdetails', monsterDetails)
+  } else if (localStorage.getItem('monsterdetails')) {
+    setMonsterDetails(JSON.parse(localStorage.getItem('monsterdetails')))
+  }
+  console.log(monsterDetails)
```
Comment (macfrei) [inline src/components/AllMonstersList.js]: Before merging, be sure to get rid of all console.logs to have your code as clean as possible.

File: src/components/OneMonsterItem.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import React from 'react'
+import styled from 'styled-components'
```
Comment (macfrei) [inline src/components/OneMonsterItem.js:2]: For better readability you could use `'styled-components/macro'`

**Review by macfrei — COMMENTED** [review]
Just minor comments, the code in general looks good. Keep up the good work 👍

File: src/components/OneMonsterItem.js (hunk 3 above)
Comment (janhenrikfock) [reply src/components/OneMonsterItem.js:8]: Challenge-Rating: Its a Stat provided by the object "monster". This container includes the information for the props "type" and "challenge-rating".

Maybe i should rename the ContainerTypeCR to "ContainerForTypeAndCR" ?

**Comment by janhenrikfock** [comment]
> Good Job,
> found a few little things and have one question: You named your component AllMonsterList <- LIST, but using not once the corresponding semantic tag in your code. Is this intentional?

Right. I could rename the Component to "AllMonsters" actually. That makes a lot more sense. Thanks

File: src/components/AllMonstersList.js (hunk 2 above)
Comment (janhenrikfock) [reply src/components/AllMonstersList.js]: Those are backticks ``. The fetch does require them. I just tried ;)

File: src/components/AllMonstersList.js (hunk 1 above)
Comment (janhenrikfock) [reply src/components/AllMonstersList.js]: Good spot! In my Case the api provides the name of the monster in the index, but without any caps and special signs like - or .
For Ex: 
Name: Aboleth
index: aboleth
So in this case it works, but i made a note of that for the future.

**Pull request merged**